src/functions/functions-api.ts

```diff
@@ -58,6 +60,40 @@ export type TaskOptions = DeliverySchedule & TaskOptionsExperimental & {
    * The default is 10 minutes. The deadline must be in the range of 15 seconds and 30 minutes.
    */
   dispatchDeadlineSeconds?: number;
+
+  /**
+   * The id to use for the enqueued event.
+   * If not provided, one will be automatically generated.
+   * If provided explicitly specifying a task ID enables task de-duplication. If a task's ID is
+   * identical to that of an existing task or a task that was deleted or executed recently then
+   * the call will throw a TaskAlreadyExists error. Another task with the same id can't be
```

Comment:
Suggest `TaskAlreadyExists` and capped "ID" in the next line.

Reply:
Actually wanted to get your thoughts on how to document this — instead of throwing a TaskAlreadyExists error, we changed the behavior to throw a FirebaseFunctionsError with code "functions/task-already-exists". So if a user wanted to check if a thrown error is due to the task already existing, they would check like so:

```
try {
  await queue.enqueue(...);
} catch (err: unknown) {
  if (err.code === "functions/task-already-exists") {
    ...
  }
}
```
Is this comment in the functions API sufficient?